src/RazorSlices/RazorSlice.StringBuilderExtensions.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Text;
+using System.Text.Encodings.Web;
+
+namespace RazorSlices;
+
+/// <summary>
+/// Extension methods for <see cref="RazorSlice" /> that allow rendering to a <see cref="StringBuilder" /> or string.
+/// </summary>
+public static class RazorSliceStringBuilderExtensions
+{
+    /// <summary>
+    /// Renders the template to the specified <see cref="StringBuilder"/>.
+    /// </summary>
+    /// <param name="slice">The <see cref="RazorSlice" /> instance.</param>
+    /// <param name="stringBuilder">The <see cref="StringBuilder"/> to render the template to.</param>
+    /// <param name="htmlEncoder">An optional <see cref="HtmlEncoder"/> instance to use when rendering the template. If none is specified, <see cref="HtmlEncoder.Default"/> will be used.</param>
+    /// <returns>A <see cref="ValueTask"/> representing the rendering of the template.</returns>
+    public static ValueTask RenderAsync(this RazorSlice slice, StringBuilder stringBuilder, HtmlEncoder? htmlEncoder = null)
+    {
+        using var stringWriter = new StringWriter(stringBuilder);
+        return slice.RenderAsync(stringWriter, htmlEncoder);
+    }
+
+    /// <summary>
+    /// Renders the template to a string.
+    /// </summary>
+    /// <param name="slice">The <see cref="RazorSlice" /> instance.</param>
+    /// <param name="htmlEncoder">An optional <see cref="HtmlEncoder"/> instance to use when rendering the template. If none is specified, <see cref="HtmlEncoder.Default"/> will be used.</param>
+    /// <returns>A string of the template.</returns>
+    public static string Render(this RazorSlice slice, HtmlEncoder? htmlEncoder = null)
+    {
+        var sb = new StringBuilder();
+        var task = slice.RenderAsync(sb, htmlEncoder);
+
+        if (task.IsCompletedSuccessfully)
+        {
+            task.GetAwaiter().GetResult();
+        }
+
+        return sb.ToString();
```

Comment:
This method should still be async and return `ValueTask`. If the task hasn't completed you'll need to return the result of calling a private helper method to do the actual `await` call, e.g.:

```suggestion
    public static ValueTask<string> RenderAsync(this RazorSlice slice, HtmlEncoder? htmlEncoder = null)
    {
        var sb = new StringBuilder();
        var task = slice.RenderAsync(sb, htmlEncoder);

        if (task.IsCompletedSuccessfully)
        {
            task.GetAwaiter().GetResult();
            return sb.ToString();
        }

        return AwaitRenderTask(task);
    }
    
    private static async ValueTask<string> AwaitRenderTask(ValueTask<string> renderTask)
    {
        return await renderTask;
```